contracts/dnr/mock_dnr_test.go

```diff
@@ -58,27 +57,52 @@ var _ = Describe("Dark nodes", func() {
 		BeforeEach(func() {
 			mu.Lock()
 
-			MockDarkNodeRegistrar, err = dnr.NewMockDarkNodeRegistrar()
-			MockDarkNodeRegistrar.Epoch()
+			var err error
+			DNR, err = dnr.TestnetDNR(nil)
+			Ω(err).ShouldNot(HaveOccurred())
+
+			DNR.WaitForEpoch()
 
 			configs = make([]*node.Config, NumberOfTestNODES)
 			ethAddresses = make([]*bind.TransactOpts, NumberOfTestNODES)
 			nodes = make([]*node.DarkNode, NumberOfTestNODES)
 
 			for i := 0; i < NumberOfTestNODES; i++ {
 
+				bond := stackint.FromUint(10)
+				// mul := stackint.FromUint(20)
+				// bond = bond.Exp(&mul)
 				configs[i] = MockConfig()
-				MockDarkNodeRegistrar.Register(
+				auth := bind.NewKeyedTransactor(configs[i].EthereumKey.PrivateKey)
+
+				dnr, err := dnr.TestnetDNR(auth)
+				Ω(err).ShouldNot(HaveOccurred())
+
+				err = dnr.ApproveRen(&bond)
+				Ω(err).ShouldNot(HaveOccurred())
+
+				dnr.SetGasLimit(300000)
```

Comment:
Ganache seems to have issues with gas estimation. The transaction runs fine as long as it has a high enough gas limit. This should only be used in tests, not code that will run on a real Ethereum chain. The gas limit is set to 0 afterwards, which re-enabled gas estimation.